zebra/zebra_rib.c

```diff
@@ -3246,8 +3246,10 @@ void rib_close_table(struct route_table *table)
 			if (info->safi == SAFI_UNICAST)
 				hook_call(rib_update, rn, NULL);
 
-			if (!RIB_SYSTEM_ROUTE(dest->selected_fib))
+			if (!RIB_SYSTEM_ROUTE(dest->selected_fib)) {
```

Comment:
While this is a point fix, I would prefer that we don't call the table close multiple times.  We need to fix our shutdown path to be run once.

Reply:
Fair enough - the first pass occurs during vrf_terminate(), and the second pass was introduced with the zebra_router code. it's not clear to me which of those should _stop_ calling rib_close_table()?